modules/gdscript/gdscript_parser.cpp

```diff
@@ -2486,13 +2486,16 @@ GDScriptParser::ExpressionNode *GDScriptParser::parse_call(ExpressionNode *p_pre
 		}
 	}
 
-	if (!check(GDScriptTokenizer::Token::PARENTHESIS_CLOSE)) {
+	if (previous.type == GDScriptTokenizer::Token::PARENTHESIS_OPEN) {
```

Comment:
I guess this still works since the check is done in the loop. But in this case I believe this will always be true since it's the token that makes this function be called. In this case I think we can remove the `if`.

Reply:
I thought about that, but the `GDScriptTokenizer::Token::SUPER` token also triggers this function so if we have `super.call|` for example, the condition is `false`. Because `consume(...PARENTHESIS_OPEN...` doesn't exit if `false`.
https://github.com/godotengine/godot/blob/9185b34890141b4b96a9fbb1500de35e9ca0d356/modules/gdscript/gdscript_parser.cpp#L2456-L2463

Reply:
Yeah, but at this point the `super` is already handled. It doesn't exit on this `consume` to keep parsing and detect other potential errors, so you can assume the parenthesis is there.

Reply:
Okay, I fixed it.